platform/podjs/src/browserPod.ts

```diff
@@ -15,68 +15,130 @@ import * as zip from "@zip.js/zip.js";
 
-class LocalStoragePolyIn implements PolyIn {
-    private static readonly storageKey = "polyInStore";
-    private store = JSON.parse(
-        localStorage.getItem(LocalStoragePolyIn.storageKey) || "[]"
-    );
+async function migrateData(db: IDBDatabase): Promise<IDBDatabase> {
+    const polyInData = localStorage.getItem("polyInStore");
+    const polyOutData = localStorage.getItem("files");
+
+    if (polyInData || polyOutData) {
+        const quads = polyInData ? JSON.parse(polyInData) : [];
+        const files: FileInfo[] = [];
+
+        if (polyOutData) {
+            for (const [, file] of JSON.parse(polyOutData)) {
+                const dataUrl = localStorage.getItem(file.id);
+                if (dataUrl) {
+                    files.push({
+                        id: file.id,
+                        name: file.name,
+                        time: file.time,
+                        blob: await (await fetch(dataUrl)).blob(),
+                    });
+                }
+            }
+        }
 
-    async match(matcher: Partial<Matcher>): Promise<RDF.Quad[]> {
-        return this.store.filter((quad: RDF.Quad) => {
-            if (matcher.subject && quad.subject.value != matcher.subject.value)
-                return false;
-            if (matcher.object && quad.object.value != matcher.object.value)
-                return false;
-            if (
-                matcher.predicate &&
-                quad.predicate.value != matcher.predicate.value
-            )
-                return false;
-            return true;
+        await new Promise((resolve, reject) => {
+            const storeNames = [OBJECT_STORE_POLY_IN, OBJECT_STORE_POLY_OUT];
+            const tx = db.transaction(storeNames, "readwrite");
+            const polyInStore = tx.objectStore(OBJECT_STORE_POLY_IN);
+            for (const quad of quads) polyInStore.add(quad);
+            const polyOutStore = tx.objectStore(OBJECT_STORE_POLY_OUT);
+            for (const file of files) polyOutStore.add(file);
+            tx.oncomplete = () => resolve(null);
+            tx.onerror = tx.onabort = () => reject(tx.error);
         });
+
+        localStorage.clear();
     }
 
-    async select(matcher: Partial<Matcher>): Promise<RDF.Quad[]> {
-        return this.store.filter((quad: RDF.Quad) => {
-            if (!quad) return false;
-            if (matcher.subject && quad.subject.value != matcher.subject.value)
-                return false;
-            if (matcher.object && quad.object.value != matcher.object.value)
-                return false;
-            if (
-                matcher.predicate &&
-                quad.predicate.value != matcher.predicate.value
-            )
-                return false;
-            return true;
+    return db;
+}
+
+function openDatabase(): Promise<IDBDatabase> {
+    return new Promise((resolve, reject) => {
+        const request = indexedDB.open(DB_NAME, DB_VERSION);
+
+        request.onupgradeneeded = () => {
```

Comment:
When will this event occur? When are we changing the database version number?

Reply:
The `onupgradeneeded` listener is invoked if the database does not exist (e.g. on first run, or after browsing data have been cleared), and when the version passed to `indexedDB.open` changed. The latter is not a scenario we have until we change the database layout after this change lands.

Reply:
Again, it would be convenient if the database layout is documented. We just have a couple of keys we can see here and there.

Reply:
> Hence the layout of the `poly-in` object store is simply `RDF.Quad`, and for `poly-out` it's `FileInfo`.

That sentence would look great on a docstring.